src/dune_pkg/opam_solver.mli

```diff
@@ -0,0 +1,20 @@
+open Stdune
+
+module Context_for_dune : sig
+  include Opam_0install.S.CONTEXT
+
+  val create_dir_context :
+       solver_env:Solver_env.t
+    -> env:(string -> OpamVariable.variable_contents option)
```

Comment:
Not entirely related, but I would prefer this naked `string` type turned into something abstract.

Reply:
I'm planning to remove this argument soon when I move all environment variables inside `Solver_env.t` so I'll leave it for now.